build.gradle.kts

```diff
@@ -128,6 +130,13 @@ tasks.test {
     setForkEvery(1)
 }
 
+androidStudioTests {
+    val autoDownloadAndRunInHeadless = providers.gradleProperty("autoDownloadAndRunInHeadless").orNull == "true"
```

Comment:
How about you use `map` here, so you can convert to a Property<Boolean>` and use that directly. That would mean you don't need to evaluate the property at configuration time.